src/aind_metadata_mapper/fip/mapper.py

```diff
@@ -0,0 +1,811 @@
+        Optional[int]
+            Fiber index if parseable, None otherwise.
+        """
+        if not fiber_name.startswith(f"{FIBER_PREFIX}_"):
+            return None
+        try:
+            return int(fiber_name.split("_")[1])
+        except (IndexError, ValueError):
+            return None
+
+    def _parse_implanted_fibers(self, subject_id: str) -> Optional[List[int]]:
+        """Parse implanted fiber indices from procedures data.
+
+        Determines which fibers were actually implanted during surgery.
+        This prevents creating patch cord connections to non-existent implanted fibers.
+
+        Parameters
+        ----------
+        subject_id : str
+            Subject ID to query.
+
+        Returns
+        -------
+        Optional[List[int]]
+            List of implanted fiber indices (e.g., [0, 1, 2] for Fiber_0, Fiber_1, Fiber_2).
+            Returns None if request fails or no fiber probes found.
+        """
+        data = get_procedures(subject_id)
+        if not data:
+            return None
+
+        implanted_indices = set()
+
+        for subject_proc in data.get("subject_procedures", []):
+            if subject_proc.get("object_type") == "Surgery":
+                for proc in subject_proc.get("procedures", []):
+                    if proc.get("object_type") == "Probe implant":
+                        implanted_device = proc.get("implanted_device", {})
+                        if implanted_device.get("object_type") == "Fiber probe":
+                            fiber_name = implanted_device.get("name", "")
+                            fiber_idx = self._extract_fiber_index(fiber_name)
+                            if fiber_idx is not None:
+                                implanted_indices.add(fiber_idx)
+
+        return sorted(list(implanted_indices)) if implanted_indices else None
+
+    def transform(self, metadata: dict) -> Acquisition:
+        """Transforms intermediate metadata into a complete Acquisition model.
+
+        Parameters
+        ----------
+        metadata : dict
+            Metadata extracted from FIP files via the extractor.
+            Must conform to the ProtoAcquisitionDataSchema JSON schema.
+
+        Returns
+        -------
+        Acquisition
+            Fully composed acquisition model.
+
+        Raises
+        ------
+        ValueError
+            If metadata validation fails.
+        """
+        # Validate against JSON schema
+        _validate_fip_metadata(metadata)
+        return self._transform(metadata)
+
+    def _transform(self, metadata: dict) -> Acquisition:
+        """Internal transform method.
+
+        Parameters
+        ----------
+        metadata : dict
+            Validated intermediate metadata dictionary.
+
+        Returns
+        -------
+        Acquisition
+            Complete acquisition metadata.
+        """
+        # Extract fields from nested structure
+        session = metadata["session"]
+        rig = metadata["rig"]
```

Comment:
If we are moving towards the instrument.json being moved to a folder, I think mapping the `rig` should be optional